### PR TITLE
Allow Sphinx version 6 and add dependency for sphinxcontrib-jquery

### PR DESCRIPTION
**Pull request by benjaoming**
[#1253](https://github.com/readthedocs/sphinx_rtd_theme/issues/1253)

~blocked by: #1381 and #1304~ Edit: no longer blocked

### CODE REVIEW

**Comment by benjaoming** [comment]
setup.cfg uses tabs and not spaces :scream:

**Comment by benjaoming** [comment]
@agjohnson if there are any problems caused by this upgrade, let's open up separate issues?

**Comment by benjaoming** [comment]
> py310-sphinx60 run-test: commands[1] | sphinx-build -b html -Dhtml4_writer=0 -d /home/circleci/project/.tox/py310-sphinx60/tmp/doctrees docs/ /home/circleci/project/.tox/py310-sphinx60/tmp/html
Running Sphinx v6.0.0b2

Checks out.

Also seems like `latest` is now mapping to Sphinx 5.3: https://www.sphinx-doc.org/en/master/changes.html#release-5-3-0-released-oct-16-2022

**Comment by polyzen** [comment]
Sphinx 6.0.0 was released today.

**Comment by benjaoming** [comment]
I defer to @agjohnson for a review here and in #1395

File: docs/requirements.txt [hunk 1]
```diff
@@ -1 +1 @@
-sphinx>=3.0
+sphinx==6.0b2
```
Comment (benjaoming) [inline docs/requirements.txt:1]:
```suggestion
sphinx==6.0.0
```

File: tox.ini [hunk 2]
```diff
@@ -35,6 +37,7 @@ deps =
     sphinx50: Sphinx>=5.0,<5.1
     sphinx51: Sphinx>=5.1,<5.2
     sphinx52: Sphinx>=5.2,<5.3
+    sphinx60: Sphinx==6.0b2
```
Comment (benjaoming) [inline tox.ini:40]:
```suggestion
    sphinx60: Sphinx==6.0.0
```

**Comment by benjaoming** [comment]
Testing and merging this for another pre-release, so we have some kind of support out...

**Pull request merged**